elevation_mapping/src/ElevationMapping.cpp

```diff
@@ -433,33 +442,59 @@ bool ElevationMapping::updateMapLocation()
+                                          std::vector<std::string>& layers, bool useRaw, bool& isSuccess)
 {
-  grid_map::Position requestedSubmapPosition(request.position_x, request.position_y);
-  Length requestedSubmapLength(request.length_x, request.length_y);
-  ROS_DEBUG("Elevation submap request: Position x=%f, y=%f, Length x=%f, y=%f.", requestedSubmapPosition.x(), requestedSubmapPosition.y(), requestedSubmapLength(0), requestedSubmapLength(1));
-  boost::recursive_mutex::scoped_lock scopedLock(map_.getFusedDataMutex());
-  map_.fuseArea(requestedSubmapPosition, requestedSubmapLength);
-
-  bool isSuccess;
   Index index;
-  GridMap subMap = map_.getFusedGridMap().getSubmap(requestedSubmapPosition, requestedSubmapLength, index, isSuccess);
-  scopedLock.unlock();
+  GridMap subMap;
+  if (useRaw) { 
+    boost::recursive_mutex::scoped_lock rawMapLock(map_.getRawDataMutex());
+    subMap = map_.getRawGridMap().getSubmap(position, length, index, isSuccess);
+    rawMapLock.unlock();
+  } else {
+    boost::recursive_mutex::scoped_lock fusedMapLock(map_.getFusedDataMutex());
+    subMap = map_.getFusedGridMap().getSubmap(position, length, index, isSuccess);
+    fusedMapLock.unlock();
+  }
 
-  if (request.layers.empty()) {
-    GridMapRosConverter::toMessage(subMap, response.map);
+  grid_map_msgs::GridMap message;
+  if (layers.empty()) {
+    GridMapRosConverter::toMessage(subMap, message);
   } else {
-    vector<string> layers;
-    for (const auto& layer : request.layers) {
-      layers.push_back(layer);
-    }
-    GridMapRosConverter::toMessage(subMap, layers, response.map);
+    GridMapRosConverter::toMessage(subMap, layers, message);
   }
+  return message;
+}
+
+bool ElevationMapping::getSubmap(grid_map_msgs::GetGridMap::Request& request, grid_map_msgs::GetGridMap::Response& response)
+{
+  ROS_DEBUG("Elevation submap request: Position x=%f, y=%f, Length x=%f, y=%f.",
+      request.position_x, request.position_y, request.length_x, request.length_y);
+
+  grid_map::Position position(request.position_x, request.position_y);
+  grid_map::Length length(request.length_x, request.length_y);
+  bool isSuccess;
+  bool useRaw = false;
+  response.map = getSubmapMessage(position, length, request.layers, useRaw, isSuccess);
 
   ROS_DEBUG("Elevation submap responded with timestamp %f.", map_.getTimeOfLastFusion().toSec());
   return isSuccess;
 }
 
+bool ElevationMapping::getRawSubmap(grid_map_msgs::GetGridMap::Request& request, grid_map_msgs::GetGridMap::Response& response)
+{
+  ROS_DEBUG("Elevation raw submap request: Position x=%f, y=%f, Length x=%f, y=%f.",
+      request.position_x, request.position_y, request.length_x, request.length_y);
+
+  grid_map::Position position(request.position_x, request.position_y);
+  grid_map::Length length(request.length_x, request.length_y);
+  bool isSuccess;
+  bool useRaw = true;
+  response.map = getSubmapMessage(position, length, request.layers, useRaw, isSuccess);
+
```

Comment:
I would add the same ROS_DEBUG output here as in the getSubmap method

Reply:
As you said before, 
`Why do you output the fusion time stamp here when getting the raw map?`
I removed the ROS_DEBUG output here.

Reply:
Ok, if outputting the time stamp is the only purpose of the log, you can leave it away.